test/02-generate.js

```diff
@@ -31,6 +33,29 @@ describe('Module - generate', function () {
         }, assertCritical(target, expected, done));
     });
 
+    it('should generate critical-path CSS from CSS files passed as Vinyl objects', function (done) {
+        var expected = read('expected/generate-default.css');
+        var target = path.resolve('.critical.css');
+        var stylesheets = ['fixtures/styles/main.css', 'fixtures/styles/bootstrap.css']
+            .map(function (filePath) {
+                return new Vinyl({
+                    cwd: '/',
+                    base: '/fixtures/',
+                    path: filePath,
+                    contents: Buffer.from(fs.readFileSync(path.join(__dirname, filePath), 'utf8'), 'utf8')
+                });
+            });
+
+        critical.generate({
+            base: 'fixtures/',
+            src: 'generate-default-nostyle.html',
```

Comment:
Uses the `-nostyle` template to make sure that styles are being loaded only from the provided Vinyl objects, and not from css files on disk.